src/content/docs/kubernetes-pixie/kubernetes-integration/advanced-configuration/configure-control-plane-monitoring.mdx

```diff
@@ -0,0 +1,499 @@
+#### Custom autodiscovery [#autodiscovery-default]
+
+Selectors used for autodiscovery are completely exposed as configuration entries in the `values.yaml` file, which means they can be tweaked or replaced to fit almost any environment where the control plane is run as a part of the cluster.
+
+An autodiscovery section looks like the following:
+
+```yaml
+autodiscover:
+  - selector: "tier=control-plane,component=etcd"
+    namespace: kube-system
+    # Set to true to consider only pods sharing the node with the scraper pod.
+    # This should be set to `true` if Kind is Daemonset, `false` otherwise.
+    matchNode: true
+    # Try to reach etcd using the following endpoints.
+    endpoints:
+      - url: https://localhost:4001
+        insecureSkipVerify: true
+        auth:
+          type: bearer
+      - url: http://localhost:2381
+  - selector: "k8s-app=etcd-manager-main"
+    namespace: kube-system
+    matchNode: true
+    endpoints:
+      - url: https://localhost:4001
+        insecureSkipVerify: true
+        auth:
+          type: bearer
+```
+
+The `autodiscover` section contains a list of autodiscovery entries. Each entry has:
+* `selector`: A string-encoded label selector that will be used to look for pods.
+* `matchNode`: If set to true, it will additionally limit discovery to pods running in the same node as the particular instance of the DaemonSet performing discovery.
+* `endpoints`: A list of endpoints to try if a pod is found for the specified selector.
+
+Additionally, each `endpoint` has:
+* `url`: URL to target, including scheme. Can be `http` or `https`.
+* `insecureSkipVerify`: If set to true, certificate will not be checked for `https` URLs.
+* `auth.type`: Which mechanism to use to authenticate the request. Currently, the following methods are supported:
+ * None: If `auth` is not specified, the request will not contain any authentication whatsoever.
+ * `bearer`: The same bearer token used to authenticate against the Kubernetes API will be sent to this request.
+ * `mtls`: mTLS will be used to perform the request.
+
+##### mTLS [#mtls]
+
+For the `mtls` type, the following needs to be specified:
+
+```yaml
+endpoints:
+  - url: https://localhost:4001
+    auth:
+      type: mtls
+      mtls:
+        secretName: secret-name
+        secretNamespace: secret-namespace
+```
+
+Where `secret-name` is the name of a Kubernetes TLS Secret, which lives in `secret-namespace`, and contains the certificate, key, and CA required to connect to that particular endpoint.
+
+The integration fetches this secret in runtime rather than mounting it, which means it requires an RBAC role granting it access to it. Our Helm Chart automatically detects `auth.mtls` entries at render time and will automatically create entries for these particular secrets and namespaces for you, unless `rbac.create` is set to false.
+
+Our integration accepts a secret with the following keys:
+- `cert`: The PEM-encoded certificate that will be presented to etcd
```

Comment:
this changed I believe

Reply:
The old behavior is still there, I'll add the new one when we merge (and test) the TLS type certs. For now I'd rather not block this PR until that happens 😓

Reply:
Yeah, but I would assume that is merged at some point. I would be bold and assume new behaviour already, but your call